src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -89,7 +89,8 @@ class RelayDefaultNetworkLayer {
         }
       }).catch(
         error => request.reject(error)
-      )
+      ),
+      request
```

Comment:
The code as implemented is equivalent to `Promise.all(requests.map(req => (foo, req)))` which is equivalent to `Promise.all(requests.map(req => req))` - in other words, the Promise.all does not wait for the results of `_sendQuery` or any of the chained promises. The default network layer intentionally does not provide access to the underlying payloads (note plural: there is no single `response`). 

I would recommend implementing a custom network layer instead.